### PR TITLE
RecipeRepository: Add `GetByName` and `GetRecipeGroupID`

### PR DESCRIPTION
**Pull request by Zouizoui78**

### CODE REVIEW

File: backend/internal/repository/recipe_test.go [hunk 1]
```diff
@@ -220,16 +220,71 @@ func TestGetByID(t *testing.T) {
...
+			"",
+			"Ch",
+			false,
+			[]model.Recipe{
+				testRecipes[1], testRecipes[0],
+			},
+			nil,
+		},
+	}
+
```
Comment (popobg) [inline backend/internal/repository/recipe_test.go:257]: You didn't name you test cases.
+ maybe you can add a test with a string that matches nothing and returns an empty string?
+ the field "err" is always nil, so we can remove it from the struct tests.

File: backend/internal/repository/recipe_test.go [hunk 2]
```diff
@@ -281,6 +346,56 @@ func TestGetByGroupID(t *testing.T) {
...
+		{
+			name:     "valid id",
+			id:       1,
+			expected: 1,
+		},
+		{
+			name:     "valid id 2",
+			id:       3,
+			expected: 2,
+		},
```
Comment (popobg) [inline backend/internal/repository/recipe_test.go]: I think one valid test should suffice.

**Review by popobg — CHANGES_REQUESTED** [review]

**Review by popobg — APPROVED** [review]
